Agent clean up  - remove deprecated modules

felipegonzalezmv opened the pull request:
### Description

This PR removes three deprecated agent modules (OpenSCAP, CIS-CAT, and Osquery) from the Wazuh Dashboard plugins as part of the Wazuh 5.0 agent clean-up initiative. The removal includes complete elimination of UI components, API endpoints, sample data generation, data fields, visualizations, testing infrastructure, Docker configurations, and related documentation across the entire codebase.

The changes ensure no orphaned code or references remain, providing a clean foundation for Wazuh 5.0 while maintaining backward compatibility for existing functionality.

### Issues Resolved

- #7642 
- Resolves #7632 - Agent clean-up objective: Remove deprecated modules
- Addresses functional requirement from wazuh/wazuh#30435 - Remove deprecated agent modules

### Evidence

**Modules Removed:**

- **OpenSCAP Module**: Complete removal of UI, APIs, sample data, tests, and Docker configs
- **CIS-CAT Module**: Complete removal of UI, APIs, sample data, tests, and Docker configs
- **OSQuery Module**: Complete removal of UI, APIs, sample data, tests, Docker configs, and integrity monitoring references

<img width="1917" height="936" alt="image" src="https://github.com/user-attachments/assets/d13f2c00-9cf7-48f8-a54f-aa7a67d7185c" />


**Files Affected:** 100+ files across UI components, backend services, testing infrastructure, and Docker configurations

### Test

**Manual Testing:**

1. Start the Wazuh Dashboard
2. Navigate to **Server Management > Settings**
3. Verify that OpenSCAP, CIS-CAT, and OSQuery modules are no longer listed
4. Check **Server Management > Configuration** sections to ensure deprecated module configuration pages are removed
5. Verify no console errors related to missing components

**Verification Commands:**

### Check List

### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

lchico commented:
@felipegonzalezmv, please note that Azure will not be removed.

For more information, please refer to the research issue:

https://github.com/wazuh/wazuh/issues/31123

felipegonzalezmv commented:
# Update - 08/12/2025

## Azure Logs Module Restoration

### Summary

The **Azure Logs module** has been successfully restored to the Wazuh Dashboard Plugins repository. This module was previously removed during the Wazuh 5.0 cleanup but has now been reinstated.

### Changes Made

- Restored Azure Logs UI components and configuration pages
- Re-enabled Azure Logs API endpoints and data processing
- Updated Office365 integration to include Azure Active Directory events
- Fixed all related tests and snapshots

### Module Status

- ✅ **Azure Logs**: Fully restored and functional
- ❌ **OpenSCAP**: Remains removed
- ❌ **CIS-CAT**: Remains removed
- ❌ **OSQuery**: Remains removed

### Access

The Azure Logs module is accessible via:
**Server Management → Settings → Cloud Security → Azure**

felipegonzalezmv commented on plugins/main/public/controllers/management/components/management/configuration/configuration-switch.js:
```diff
@@ -27,10 +27,9 @@ import WzConfigurationClientBuffer from './client-buffer/client-buffer';
+
 import WzConfigurationVulnerabilities from './vulnerabilities/vulnerabilities';
-import WzConfigurationOsquery from './osquery/osquery';
+
```
Resolve in this commit https://github.com/wazuh/wazuh-dashboard-plugins/commit/eca6397bffb61d03179948646037003bd6edec0f

gonzaarancibia reviewed (changes requested):
🔴 CR

I found code related to the `CIS-CAT` module in the project.

**Some locations where it appears:**

- plugins/main/common/api-info/endpoints.json
- plugins/main/common/api-info/security-actions.json
- plugins/wazuh-core/common/api-info/endpoints.json
- plugins/main/server/lib/reporting/agent-configuration.ts

Screenshots:

<img width="889" height="588" alt="Image" src="https://github.com/user-attachments/assets/3d10df8e-7cd6-4626-aa0b-e90f243a8a39" />

<img width="580" height="1113" alt="Image" src="https://github.com/user-attachments/assets/b6249804-02db-498e-affa-be6b2a0da091" />

<img width="487" height="1087" alt="Image" src="https://github.com/user-attachments/assets/af872305-cea7-4df2-b4c9-33ade0d95ffd" />

felipegonzalezmv commented:
## Update

### Endpoints Updated to Version 5.0

Updated all API endpoints to use Wazuh API specification version 5.0 from the upstream source.

### Fixed Missing Descriptions

Applied a small fix to handle endpoints that don't include a `description` field in the API specification. The generation script now properly handles cases where the description property is missing or undefined.

### Removed Unused Files from wazuh-core

Deleted `endpoints.json` and `security-actions.json` files from the `wazuh-core` plugin directory since they are not actually used there. These files are only utilized in `plugins/main` where they are properly generated and maintained.

### File Locations

- **Used**: `plugins/main/common/api-info/endpoints.json`
- **Used**: `plugins/main/common/api-info/security-actions.json`
- **Removed**: `plugins/wazuh-core/common/api-info/endpoints.json` ❌
- **Removed**: `plugins/wazuh-core/common/api-info/security-actions.json` ❌

gonzaarancibia reviewed (dismissed):
## Test 🟡 

I found code related to the CIS-CAT module in the project. If this is one of the modules that can't be removed manually, please discard my suggestion.

<details><summary>Evidence</summary>
<p>


<img width="539" height="805" alt="Image" src="https://github.com/user-attachments/assets/9dfc2593-ae1c-4bb5-bccc-3f0f8fa4b2a2" />
<img width="1153" height="888" alt="Image" src="https://github.com/user-attachments/assets/e09fda02-09d3-4252-9ec1-44f790c3dc9a" />

</p>
</details>

gonzaarancibia reviewed (dismissed):
🟢 LGTM
🟢 CR

<img width="1580" height="1080" alt="Image" src="https://github.com/user-attachments/assets/c4c2a108-1c4b-49d4-b2f5-64ed9106d2a4" />

guidomodarelli reviewed (dismissed):
LGTM! 🟢

gonzaarancibia reviewed (dismissed):
🟢 LGTM
🟢 CR

<img width="1788" height="1116" alt="Image" src="https://github.com/user-attachments/assets/4b02450e-bdf3-4ef2-9808-d40b5239ffa5" />

Desvelao commented on CHANGELOG.md:
```diff
@@ -21,6 +21,7 @@ All notable changes to the Wazuh app project will be documented in this file.
 - Removed the settings related to monitoring and statistics job from the configuration [#7597](https://github.com/wazuh/wazuh-dashboard-plugins/pull/7597)
 - Removed prompt related to statistic job is disabled in Statistics app [#7597](https://github.com/wazuh/wazuh-dashboard-plugins/pull/7597)
 - Removed the configuration for modules that relied on the following deprecated daemons: wazuh-agentlessd, wazuh-csyslogd, wazuh-dbd, wazuh-integratord, wazuh-maild, and wazuh-reportd. [#7612](https://github.com/wazuh/wazuh-dashboard-plugins/pull/7612)
+- Removed deprecated agent modules OpenSCAP, CIS-CAT, OSQuery [#7645](https://github.com/wazuh/wazuh-dashboard-plugins/pull/7645)
```
```suggestion
- Removed deprecated modules OpenSCAP, CIS-CAT, Osquery [#7645](https://github.com/wazuh/wazuh-dashboard-plugins/pull/7645)
```

Desvelao commented on plugins/main/common/constants.ts:
```diff
@@ -484,16 +471,13 @@ export enum WAZUH_MODULES_ID {
   POLICY_MONITORING = 'pm',
   SECURITY_CONFIGURATION_ASSESSMENT = 'sca',
   AUDITING = 'audit',
-  OPEN_SCAP = 'oscap',
```
**nitpick**: Remove the usage of this variable in `plugins/main/public/utils/components-os-support.js`

gonzaarancibia reviewed (approved):
🟢 LGTM

guidomodarelli reviewed (approved):
LGTM! 🟢
